[Blog Post] Introducing Agentic Search UI in OpenSearch (#4048)
opensearch-project/project-website: _posts/2026-01-12-introducing-agentic-search-ui.md

---

We are excited to announce an all-new agentic search user experience available in OpenSearch 3.4. This interface in OpenSearch Dashboards provides a streamlined way for configuring agents, testing them using natural language queries, and exporting configurations for downstream application integration.

## What is agentic search?

_Agentic search_ transforms the way you interact with data by letting you use natural language queries instead of writing complex search syntax. An intelligent agent interprets your question, automatically plans the search, and returns relevant results, providing complete transparency into its decision-making process. For more information, read [our previous blog post](https://opensearch.org/blog/introducing-agentic-search-in-opensearch-transforming-data-interaction-through-natural-language/). For a detailed overview of agentic search capabilities and architecture, see the [agentic search documentation](https://docs.opensearch.org/latest/vector-search/ai-search/agentic-search/index/).

## Using agentic search in OpenSearch Dashboards

> To access agentic search in OpenSearch Dashboards, go to **OpenSearch Dashboards** > **OpenSearch Plugins** > **AI Search Flows** and create a new agentic search workflow. The interface has two main sections: configure agents on the left and test searches on the right. For detailed information about the interface components, agent types, available tools, and configuration options, see [Configuring agentic search](https://docs.opensearch.org/latest/vector-search/ai-search/building-agentic-search-flows/).

Reviewer: ```suggestion
To access agentic search in OpenSearch Dashboards, go to **OpenSearch Dashboards** > **OpenSearch Plugins** > **AI Search Flows** and create a new agentic search workflow. The interface has two main sections: agent configuration options on the left and search execution capabilities on the right. For detailed information about the interface components, agent types, available tools, and configuration options, see [Configuring agentic search](https://docs.opensearch.org/latest/vector-search/ai-search/building-agentic-search-flows/).
```

Next, try updating the model's available tools. To enhance the agent's flexibility and enable access to real-time information, enable the Web Search tool:

1. Under **Configure Agent** > **Tools** > **Web Search**, select **Enable**. The tool form is expanded automatically.
2. Under **Engine**, enter `duckduckgo`---this is the simplest option and doesn't require additional permissions or configuration.
3. Select **Update agent**.
4. Under **Agentic Search**, select **Search** to run a new search using the updated agent. The agent can now address questions that require information beyond the scope of your indexed dataset.

Now try a new search query that requires fetching external information and evaluate the results. Enter the query `Shoes from the brand Serena Williams wears` and select **Search**. The generated query filters specifically for the brand Nike, as shown in the following image.

![Web query](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-7-web-query.png)

The following image shows the query results.

![Web results](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-8-web-results.png)

> The agent first searched the web to identify Serena Williams' brand, Nike, and then incorporated it into the generated query, ultimately returning Nike shoes, as shown in the agent summary in the following image.

Reviewer: ```suggestion
The agent first searched the web to identify the shoe brand associated with Serena Williams, Nike, and then incorporated it into the generated query, ultimately returning Nike shoes, as shown in the agent summary in the following image.
```

![Nike](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-9-web-summary.png)

> Next, try integrating with the MCP server to let the agent view your order history for a more customized, personalized search experience. This server has a tool `simple_get_order_history` that you can use to return your order history details.

Reviewer: ```suggestion
Next, try integrating with the MCP server to let the agent view your order history for a more customized, personalized search experience. This server has a tool `simple_get_order_history` that you can use to return your order history details:
```

1. Under **Configure Agent** > **MCP Servers**, select **Add MCP server**.
2. From the **MCP Server** dropdown, select `Customer Order History MCP Server`.
3. Under **Tool filters**, enter `simple_get_order_history` to restrict the agent's access so that only this tool from the MCP server is available. This tool does not take any parameters and returns the default order history.
4. Select **Update agent**.

The following image shows the MCP server configuration.

![MCP](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-10-mcp.png)

Now try searching for athletic shorts, using the order history to find similar products and brands. Update the query to `Athletic shorts similar to my order history`. The generated query is displayed in the following image.

![Query](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-11-query.png)

The following image shows the results, which include athletic shorts from Adidas and Nike.

![Results](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-12-results.png)

> The following image shows the agent summary that demonstrates how the agent determined that Nike and Adidas brands were present in the order history and applied the corresponding brand filters in the DSL query.

Reviewer: ```suggestion
The following image shows the agent summary that demonstrates how the agent determined that the Nike and Adidas brands were present in the order history and applied the corresponding brand filters in the DSL query.
```

![Pipeline](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-17-pipeline.png)

Example query formats for integration are displayed in the following image.

![Query](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-18-query.png)

### Example 2: Fast product filtering

_Flow agents_ provide fast, single-query search capabilities, performing significantly faster than conversational agents when multi-turn conversations or complex reasoning are not required. Let's build an application that uses this efficiency.

#### Step 1: Create an agent

Create a flow agent:

1. Select **Create agent**.
2. Enter a name: `My flow agent`.
3. Change the agent type to **Flow**.
4. Under the **Query Planning** tool, select `Bedrock Claude 4.5`.
5. Select **Create**.

The following image shows the flow agent configuration form.

![Flow](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-14-flow.png)

#### Step 2: Run agentic searches

Test the agent by running a direct product search:

1. Select the `demo_amazon_fashion` index.
2. Enter a search query, for example, `Women's running shoes under $100`.
3. Select **Search**.

> The agent generates an optimized query DSL and runs the search, returning women's running shoes under $100. The following image demonstrates the flow agent's query generation.

Reviewer: ```suggestion
The agent generates optimized query DSL and runs the search, returning women's running shoes under $100. The following image demonstrates the flow agent's query generation.
```

![Flow query](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-15-flow-query.png)

The following image shows the resulting product matches.

![Flow results](/assets/media/blog-images/2026-01-12-introducing-agentic-search-ui/blog-16-flow-results.png)

#### Step 3: Tune your agent

Similarly to the previous example, you can test how your agent performs by using different models for query generation.

## Next steps

Ready to transform your search experience? Start by testing agentic search in OpenSearch Dashboards, then use the insights you gain to build powerful production applications. For more information and examples, see the following resources:

- **Experiment first**: Try agentic search on the [ML playground](https://ml.playground.opensearch.org/app/opensearch-flow#/workflows) with pre-configured agents

> - **Plan your implementation**: Review the [agentic search OpenSearch Dashboards documentation](https://docs.opensearch.org/latest/vector-search/ai-search/building-agentic-search-flows/)

Reviewer: ```suggestion
- **Plan your implementation**: Review the [agentic search OpenSearch Dashboards documentation](https://docs.opensearch.org/latest/vector-search/ai-search/building-agentic-search-flows/).
```

> - **Understand the architecture**: Explore the [agentic search documentation](https://docs.opensearch.org/latest/vector-search/ai-search/agentic-search/index/)

Reviewer: ```suggestion
- **Understand the architecture**: Explore the [agentic search documentation](https://docs.opensearch.org/latest/vector-search/ai-search/agentic-search/index/).
```

> - **Learn from examples**: Read our [previous agentic search blog post](https://opensearch.org/blog/introducing-agentic-search-in-opensearch-transforming-data-interaction-through-natural-language/)

Reviewer: ```suggestion
- **Learn from examples**: Read our [previous agentic search blog post](https://opensearch.org/blog/introducing-agentic-search-in-opensearch-transforming-data-interaction-through-natural-language/).
```
